apps/speed.c

```diff
@@ -1656,6 +1907,126 @@ int speed_main(int argc, char **argv)
         }
     }
 
+    /* find all KEMs currently available */
+    kem_stack = sk_EVP_KEM_new(kems_cmp);
+    EVP_KEM_do_all_provided(app_get0_libctx(), collect_kem, kem_stack);
+
+    kems_algs_len = 0;
+
+    for (idx = 0; idx < (unsigned int)sk_EVP_KEM_num(kem_stack); idx++) {
+        EVP_KEM *kem = sk_EVP_KEM_value(kem_stack, idx);
+
+        if (!strcmp(EVP_KEM_get0_name(kem), "RSA")) {
+            if (kems_algs_len + 7 >= MAX_KEM_NUM) {
+                BIO_printf(bio_err,
+                       "Too many KEMs registered. Change MAX_KEM_NUM.\n");
+                goto end;
+            }
+            kems_doit[kems_algs_len] = 1;
+            kems_algname[kems_algs_len++] = OPENSSL_strdup("RSA512");
```

Comment:
I don't see where these `OPENSSL_strdup` are released ?

Reply:
Thanks for the catch. Now added in https://github.com/openssl/openssl/pull/19968/commits/b7b2a70ec374b2d556e1290a0abceb48009ae939.